### DISCUSSION TITLE
Social Login field Mapping?

### DISCUSSION THREAD

**Discussion by iyobo** [logto-io/logto#4075] [General]
Most IDPs have the ability to map fields from the social login identity to the IDP.

E.g, being able to map accessTokens from Apple sign in so you can later use it to delete/disconnect users as required by the app store.

Is this a part of the plan?

**Comment by darcyYe** [comment]
Thanks for sharing your ideas!
We use social connectors to request social identity user profiles with the authentication of users. During the process, we got the access token but with the least scopes which were enough for getting the user's social identity info. The access token is used once. We do not provide a standard way to store and manage access tokens from social login right now.
But if there are solid use cases, we may consider supporting this scenario, you are more than welcome to share your use case.

**Reply by iyobo** [reply]
The use case is as I mentioned earlier.

Apple demands that to enable any kind of social login on iOS apps, you must enable Signin with Apple.

Sign in with Apple has an additional demand lining up with GDPR's Right to be forgotten. When a user deletes their account, it is requested that the service provider makes a call to the Apple API with the initial access token from Apple for that user specifically to detach that users AppleId from your service.

TLDR without this feature, this identity management platform cannot be used by any mobile apps for social login. At least not if that app intends to be put on the app store, as it will be denied.

**Reply by darcyYe** [reply]
Our team had a discussion on this recently and found that this should be important. Will handle this likely in November. Stay tunned.

**Reply by hadrien** [reply]
Any update @darcyYe ? Thanks so much

**Comment by Tito1337** [comment]
I'm also interested in this feature. Here is another use case :

Currently, social connectors receive `given_name` and `family_name` from the provider's response, but only the combined `name` field is extracted into SocialUserInfo. The individual name parts are discarded.

Meanwhile, the UserProfile schema already supports `givenName`, `familyName`, and `middleName`  : these fields just never get populated from social login.

The data is preserved in `rawData`, so the information isn't lost, but there's no built-in way to flow it into the user profile...
